nilearn/reporting/glm_reporter.py

```diff
@@ -967,3 +968,144 @@ def _dataframe_to_html(df, precision, **kwargs):
         html_table = df.to_html(**kwargs)
     html_table = html_table.replace('border="1" ', '')
     return html_table
+
+
+def _clean_contrast_name(contrast_name):
+    new_name = "".join(ch for ch in contrast_name if ch.isalnum())
+    if new_name != contrast_name:
+        warnings.warn("Contrast name '{}' changed to '{}'".format(contrast_name, new_name))
+    return new_name
+
+
+def save_glm_results(model, contrasts, out_dir=".", prefix=None):
+    """Save GLM results to BIDS-like files.
+
+    To output:
+    - design matrix
+    - contrast plot
+    - contrast-level stat maps
+    - model-level stat maps
+    - metadata
+    """
+    if isinstance(prefix, str) and not prefix.endswith("_"):
+        prefix += "_"
+    else:
+        prefix = ""
+
+    out_dir = os.path.abspath(out_dir)
+
+    # Write out design matrices to files.
+    if hasattr(model, "design_matrices_"):
+        design_matrices = model.design_matrices_
+    else:
+        design_matrices = [model.design_matrix_]
+
+    if len(design_matrices) > 1:
+        for i_run, dm in enumerate(design_matrices):
+            run_name = i_run + 1
+            dm_file = os.path.join(
+                out_dir,
+                "{}run-{}_design.tsv".format(prefix, run_name),
+            )
+            dm.to_csv(dm_file, sep="\t", index=False)
+
+            dm_fig_file = os.path.join(
+                out_dir,
+                "{}run-{}_design.svg".format(prefix, run_name),
```

Comment:
Just curious: is the use of svg recommended by BIDS ?

Reply:
Since BIDS models (and more importantly their outputs) aren't actually in the specification yet, I'm basing this off of `fitlins`, which does use SVG. Still, I don't think it's necessary to stick with their approach completely. Would you prefer another format?

Reply:
svg is fine, I was just curious.